### PR TITLE
Add an option to configure Job specific timeout values

### PR DESCRIPTION
**Pull request by maluchari**
## Summary

Current default value of timeout is 12 hours. Since how long an entire job can take depend on number of tables and also on concurreny, making the timeout configurable in this PR.

## Changes

- [ ] Client-facing API Changes
- [X] Internal API Changes
- [ ] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [X] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java [hunk 1]
```diff
@@ -119,10 +119,11 @@ public static void main(String[] args) {
             Executors.newFixedThreadPool(getNumParallelJobs(cmdLine)),
             tasksFactory,
             tablesClientFactory.create());
-    app.run(operationType, operationTaskCls.toString(), isDryRun(cmdLine));
+    app.run(operationType, operationTaskCls.toString(), isDryRun(cmdLine), getTaskTimeOut(cmdLine));
   }
 
-  protected void run(JobConf.JobTypeEnum jobType, String taskType, boolean isDryRun) {
+  protected void run(
+      JobConf.JobTypeEnum jobType, String taskType, boolean isDryRun, int taskTimeOutInHours) {
```
Comment (teamurko) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java:126]:
```suggestion
      JobConf.JobTypeEnum jobType, String taskType, boolean isDryRun, int tasksWaitHours) {
```

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java [hunk 2]
```diff
@@ -63,7 +63,7 @@
  */
 @Slf4j
 public class JobsScheduler {
-  private static final int TASKS_WAIT_TIMEOUT_HOURS = 12;
+  private static final int DEFAULT_TASKS_WAIT_TIMEOUT_HOURS = 12;
```
Comment (teamurko) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java:66]:
```suggestion
  private static final int DEFAULT_TASKS_WAIT_HOURS = 12;
```

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java [hunk 3]
```diff
@@ -290,6 +290,13 @@ protected static CommandLine parseArgs(String[] args) {
             .longOpt("dryRun")
             .desc("Dry run without actual action")
             .build());
+    options.addOption(
+        Option.builder(null)
+            .required(false)
+            .hasArg(false)
+            .longOpt("taskTimeOutInHours")
```
Comment (teamurko) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java:297]:
```suggestion
            .longOpt("tasksWaitHours")
```

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java [hunk 4]
```diff
@@ -384,4 +391,12 @@ protected static int getNumParallelJobs(CommandLine cmdLine) {
     }
     return ret;
   }
+
+  protected static int getTaskTimeOut(CommandLine cmdLine) {
+    int ret = DEFAULT_TASKS_WAIT_TIMEOUT_HOURS;
+    if (cmdLine.hasOption("taskTimeOutInHours")) {
+      ret = Integer.parseInt(cmdLine.getOptionValue("taskTimeOutInHours"));
+    }
+    return ret;
```
Comment (teamurko) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java:400]:
```suggestion
    return NumberUtils.toInt(cmdLine.getOptionValue("taskTimeOutInHours"), DEFAULT_TASKS_WAIT_HOURS);
```
Can just do it in place

**Review by teamurko — CHANGES_REQUESTED** [review]
Thank you for the change, I have minor naming suggestions

**Review by teamurko — APPROVED** [review]

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java [hunk 5]
```diff
@@ -384,4 +392,9 @@ protected static int getNumParallelJobs(CommandLine cmdLine) {
     }
     return ret;
   }
+
+  protected static int getTaskTimeOut(CommandLine cmdLine) {
+    return NumberUtils.toInt(
+        cmdLine.getOptionValue("taskTimeOutInHours"), DEFAULT_TASKS_WAIT_HOURS);
```
Comment (abhisheknath2011) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java:398]:
```suggestion
        cmdLine.getOptionValue("tasksWaitHours"), DEFAULT_TASKS_WAIT_HOURS);
```
Should this be `tasksWaitHours`?

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java (hunk 1 above)
Comment (maluchari) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java]: Sure. Updated

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java (hunk 4 above)
Comment (maluchari) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java]: Ack. Updated this method alone and leaving other methods as such

**Review by teamurko — APPROVED** [review]

**Review by abhisheknath2011 — APPROVED** [review]

**Pull request merged**